plugins/metrics/runtime/runtime.go

```diff
@@ -0,0 +1,286 @@
+package runtime // import "go.opentelemetry.io/contrib/plugins/metrics/runtime"
+
+import (
+	"context"
+	goruntime "runtime"
+	"sync"
+	"time"
+
+	"go.opentelemetry.io/otel/api/metric"
+	"go.opentelemetry.io/otel/api/unit"
+)
+
+type Runtime struct {
+	mu       sync.RWMutex
+	meter    metric.Meter
+	interval time.Duration
+	done     chan bool
+
+	metrics struct {
+		goCgoCalls metric.Int64Counter
+		goLookups  metric.Int64Counter
+		goGcCount  metric.Int64Counter
+		gcPauseNs  metric.Int64Measure
+	}
+
+	memStats    goruntime.MemStats
+	numCgoCalls int64
+}
+
+func NewRuntime(meter metric.Meter, interval time.Duration) *Runtime {
```

Comment:
Let's name this "New", since it's going to be called `runtime.New()` there's less stutter. I noticed a concern. I think the ideal case here is that all the instruments are asynchronous ("observers"), meaning the SDK controls the interval. It won't be the runtime's decision how often to snapshot instruments, so at some point the `interval` parameter might go away. Is that ok with you?

Reply:
Help me understand this. Once sdk is updated for new types and batch reporting is in, using async types for everything would be nice and clean. No mismatched ticker and the cache of the struct. 

However, since runtime.memstat metrics for go involve a stop-the-world action. It is not OK to set it arbitrarily frequent interval. So I would want either we allow an interval for how often to poll, or two metric.meter’s with different intervals.

Reply:
> I would want either we allow an interval for how often to poll...

This interval is set when a Controller is created. Asynchronous instruments observe on this cadence.

I agree that this should be linked to that interval. If it is specified here it will be the instrumenter defining this interval instead of the operator or end-user of the instrumented code.

Reply:
This function will also need a docstring.

Reply:
One of the considerations behind the asynchronous instrument API--the callback registration mechanism--is to support instruments with variable rate intervals.  So, this PR has done a fine job of demonstrating all the advanced things.

The approach here:
https://github.com/open-telemetry/opentelemetry-go/pull/634
uses a somewhat heavyweight API, and we're considering whether the semantics get confusing when we relax it a bit, please have a look.

Reply:
> we allow an interval for how often to poll

This is a bit complex, but understandable. I wonder if we should allow this code to set a minimum interval between polling, as a hard-stop on the rate of stop-the-world behavior. If an observer callback is called too frequently it can observe a cached result. 

Another option would be for the callback to return some kind of "not now" error code to say that the observer is not ready and would like to defer _without making an observation_.